libres/tests/analysis/test_save_parameters.cpp

```diff
@@ -0,0 +1,87 @@
+#include <filesystem>
+#include <iostream>
+#include <fstream>
+
+#include "catch2/catch.hpp"
+
+#include <ert/enkf/enkf_fs.hpp>
+#include <ert/analysis/update.hpp>
+#include <ert/enkf/ensemble_config.hpp>
+#include <ert/enkf/enkf_config_node.hpp>
+#include <ert/enkf/enkf_node.hpp>
+#include <ert/enkf/enkf_defaults.hpp>
+#include <ert/enkf/enkf_fs.hpp>
+#include <ert/enkf/local_ministep.hpp>
+#include <ert/util/type_vector_functions.hpp>
+
+#include "../tmpdir.hpp"
+
+extern "C" void enkf_fs_umount(enkf_fs_type *fs);
+
+TEST_CASE("Write and read a matrix to enkf_fs instance", "[analysis]") {
+    GIVEN("An instance of enkf_fs") {
+        WITH_TMPDIR;
+        // create file system
+        auto file_path = std::filesystem::current_path();
+        auto fs = enkf_fs_create_fs(file_path.c_str(), BLOCK_FS_DRIVER_ID, NULL,
+                                    true);
+
+        auto ensemble_config = ensemble_config_alloc_full("name-not-important");
+
+        // setting up a config node for a single parameter
+        auto config_node =
+            ensemble_config_add_gen_kw(ensemble_config, "TEST", false);
+        // create template file
+        std::ofstream templatefile("template");
+        templatefile << "{\n\"a\": <COEFF>\n}" << std::endl;
+        templatefile.close();
+
+        // create parameter_file
+        std::ofstream paramfile("param");
+        paramfile << "COEFF UNIFORM 0 1" << std::endl;
+        paramfile.close();
+
+        enkf_config_node_update_gen_kw(config_node, "not_important.txt",
+                                       "template", "param", nullptr, nullptr);
+
+        // creates files on fs where nodes are stored
+        enkf_node_type *node = enkf_node_alloc(config_node);
+        for (int i = 0; i < 10; i++) {
+            enkf_node_store(node, fs, {.report_step = 0, .iens = i});
+        }
+        enkf_node_free(node);
+
+        // set up ministep with one parmater and 10 realizations
+        local_ministep_type *ministep =
+            new local_ministep_type("not-important", nullptr);
+        ministep->add_active_data("TEST");
+        int_vector_type *active_index = int_vector_alloc(10, -1);
+        for (int i = 0; i < 10; i++) {
+            int_vector_iset(active_index, i, i);
+        }
+
+        WHEN("Saving parameters to enkf_fs") {
+            matrix_type *A = matrix_alloc(1, 10);
+            for (int i = 0; i < 10; i++) {
+                matrix_iset(A, 0, i, double(i) / 10.0);
+            }
+            analysis::save_parameters(fs, ensemble_config, active_index, 0,
```

Comment:
This is still the Assamlbling/Arrangment faze and thus should be in the `GIVEN` scope

Reply:
-this is still the Assambling/Arrangment faze-
That was incorrect on my part since saveing is part of the save-load mechanism that we are activating